crates/sqlbuiltins/src/functions/table/json.rs

```diff
@@ -0,0 +1,67 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use datafusion::datasource::TableProvider;
+use datafusion_ext::errors::ExtensionError;
+use datafusion_ext::functions::{FuncParamValue, TableFuncContextProvider};
+use datasources::common::url::{DatasourceUrl, DatasourceUrlType};
+use datasources::json::table::json_streaming_table;
+use datasources::object_store::generic::GenericStoreAccess;
+use protogen::metastore::types::catalog::RuntimePreference;
+
+use crate::functions::table::{table_location_and_opts, TableFunc};
+use crate::functions::{ConstBuiltinFunction, FunctionType};
+
+#[derive(Debug, Clone, Copy, Default)]
+pub struct JsonScan;
+
+impl ConstBuiltinFunction for JsonScan {
+    const NAME: &'static str = "read_json";
```

Comment:
I'm fine with just `read_json` and `read_ndjson`. This is explicit & doesn't leave any ambiguity.

Reply:
the duckdb json interface is a bit cleaner in terms of format autodetection, which this really doesn't have and that might be confusing.

If you take a newline delimited json file or glob you'll get one document per-file, with this code.

Reply:
So i think the autodetection is nice, but I think making users aware of the implications of using ndjson vs json is more important. 

We can do a lot more optimizations with ndjson than we can with json, and the code paths are very different. We had some pretty lengthy discussions over at polars when adding ndjson functionality, and ended up making them two distinct functions for this reason.

Reply:
It makes sense here for this function, and I'm willing to go with this naming, but we don't have a good answer for `CREATE EXTERNAL TABLE` which calls `ndjson` ... `json`.